File: tests/fixtures/classes/all_forms.ts
// 1. Simple class
class Animal {
    name: string;
    constructor(name: string) {
        this.name = name;
    }
    speak(): string {
        return `${this.name} makes a sound`;
    }
}

// 2. Class with extends
class Dog extends Animal {
    breed: string;
    constructor(name: string, breed: string) {
        super(name);
        this.breed = breed;
    }
    speak(): string {
        return `${this.name} barks`;
    }
}

// 3. Class with implements
interface Printable {
    print(): void;
}

class Document implements Printable {
    print(): void {
        console.log("printing...");
    }
}

// 4. Abstract class
abstract class Shape {
    abstract area(): number;
}

// 5. Exported class
export class ExportedUtil {
    static version = "1.0";
}

// 6. Default export class
export default class DefaultClass {
    value = 42;
}

File: tests/fixtures/classes/all_forms.py
class Animal:
    def __init__(self, name: str):
        self.name = name

    def speak(self) -> str:
        return f"{self.name} makes a sound"


class Dog(Animal):
    def __init__(self, name: str, breed: str):
        super().__init__(name)
        self.breed = breed

    def speak(self) -> str:
        return f"{self.name} barks"


class Utility:
    @staticmethod
    def add(a: int, b: int) -> int:
        return a + b

    @classmethod
    def create(cls) -> "Utility":
        return cls()
